### PR TITLE
Enable Queries on Imported/Exported Tables

### PR DESCRIPTION
**Pull request by dey4ss**
In hyrise/hyrise#2514, we sketched how filtering and encoding selection during import and export of tables does look like in Posgtres [1]. After a discusison in the weekly, this PR enables filters on import/export of tables. The export filter will also be implemented in Hyrise.

[1] https://www.postgresql.org/docs/15/sql-copy.html

### CODE REVIEW

File: test/queries/queries-good.sql
Comment (Bouncner) [inline test/queries/queries-good.sql:79]: Why not `file_path`?

Comment (Bouncner) [reply test/queries/queries-good.sql:79]: Besides that nothing to add.

I am nonetheless wondering, if we have tested all possible (or at least relevant) cases. But tbh, I have used queries on exports and I wasn't even aware it existed.

I guess an empty results is just a CSV file (if requested) with a header, right?

**Review by Bouncner — COMMENTED** [review]
I don't have much to say.

File: test/queries/queries-good.sql
Comment (dey4ss) [reply test/queries/queries-good.sql:79]: Just something figurative here for the example query. Can change if requested.

I just hooked the existing select statement into the export statement: ﻿https://github.com/hyrise/sql-parser/blob/396fd59aa6bd154e330eb485fee1473245a6df84/src/parser/bison_parser.y#L475
The select statement should be tested accordingly in [`select_tests.cpp`](https://github.com/hyrise/sql-parser/blob/396fd59aa6bd154e330eb485fee1473245a6df84/test/select_tests.cpp). Same goes for the `WHERE`clause of the import statement.

Last question: The Hyrise exporter will write the header regardless of the content (so an empty table results in the written header - however, we do not write CSV headers but create an extra file containing the schema information). If the desired table does not exist, we fail.

**Comment by Bouncner** [comment]
Fine from my side, but as I am not familiar with the parser, maybe wait for another approval?

File: src/sql/statements.cpp
Comment (klauck) [inline src/sql/statements.cpp:301]: Simply `free(dtable)`?

File: src/parser/bison_parser.y
Comment (klauck) [inline src/parser/bison_parser.y:39]: Why remove? Does clang complain about the format?

File: src/sql/statements.cpp
Comment (klauck) [inline src/sql/statements.cpp:361]: For me, the old version is clearer. But I'm not sure what is generally preferred.

File: src/sql/Expr.cpp
Comment (klauck) [inline src/sql/Expr.cpp:253]: see below

File: src/sql/Expr.cpp
Comment (klauck) [inline src/sql/Expr.cpp:255]: see below

**Review by klauck — APPROVED** [review]
Looks good to me.

I had only
-   one question
-   two style questions (but the proposed versions are ok for me)

.

File: src/parser/bison_parser.y
Comment (dey4ss) [reply src/parser/bison_parser.y:39]: To be honest, gonna have to look into this after that time. Think that clang-format at that time (probably 14 or so) did some weird stuff then.

File: src/sql/statements.cpp
Comment (dey4ss) [reply src/sql/statements.cpp:361]: To me, comparing a pointer `==`/`!=` `nullptr` doesn't seem right at all (not only in if clauses). Thus, the change. Maybe @Bouncner has an opinion on that

File: src/sql/statements.cpp
Comment (dey4ss) [reply src/sql/statements.cpp:301]: cppreference:
> If ptr is a null pointer, the function does nothing. 

Absolutely, gonna remove the mentioned ifs.

File: src/sql/statements.cpp
Comment (Bouncner) [reply src/sql/statements.cpp:361]: To me, comparing a pointer to `nullptr` is more obvious than casting it to another type which does implicitly the same.

Comment (dey4ss) [reply src/sql/statements.cpp:361]: Alright, I'm outvoted :D

File: src/parser/bison_parser.y
Comment (dey4ss) [reply src/parser/bison_parser.y:39]: Indeed, clang-format (15) locally screws up the indentation and even adds linebreaks in the bison control lines.

**Comment by dey4ss** [comment]
@Bouncner @klauck should be ready to go now. Knaller? 🧨 💥

**Comment by Bouncner** [comment]
> @Bouncner @klauck should be ready to go now. Knaller? 🧨 💥

🧨 💥 ❗

**Review by Bouncner — APPROVED** [review]

**Comment by dey4ss** [comment]
Argh, Travis doesn't wanna work again 😡 
Claims our credits are completely consumed while we still have lots of them, doesn't even list this branch as a PR. Starting to really hate that s***. Any ideas?

**Review by klauck — APPROVED** [review]

**Comment by Bouncner** [comment]
Mailing TravisCI? :-/

**Comment by dey4ss** [comment]
Yeah, already did that. Let's see what happens.

**Comment by dey4ss** [comment]
They fixed it.

**Pull request merged**
